keyvaluestore/nodereader.go

```diff
@@ -17,5 +19,55 @@ func (reader *NodeReaderImpl) ReadNode(nodeId uint64) (*node, error) {
 		return nil, nil
 	}
 
-	return nil, nil
+	pageSize := os.Getpagesize()
+	bytes := make([]byte, os.Getpagesize())
+	_, err := reader.file.ReadAt(bytes, int64(nodeId)*int64(pageSize))
+	if err != nil {
+		return nil, err
+	}
+
+	var node *node = &node{
+		nodeId:   nodeId,
+		n:        0,
+		keys:     make([]uint64, reader.tree.max_degree),
+		values:   make([]*[10]byte, reader.tree.max_degree),
+		children: make([]uint64, reader.tree.max_degree+1),
+		isLeaf:   false,
+		next:     0,
+		parent:   0,
+		tree:     reader.tree,
+		isDirty:  false,
+	}
+
+	node.isLeaf = ToBool(bytes[0])
+
+	if nodeId != ToUInt64(bytes[1:9]) {
```

Comment:
`nodeIdOffset`

Reply:
Since it isn't possible to save key-ranges in a variable, the result would look way worse. I left the magic numbers in.